bazel/external/boringssl_fips.genrule_cmd

```diff
@@ -119,7 +119,7 @@ rm -rf boringssl/build
 
 # Build BoringSSL.
 cd boringssl
-mkdir build && cd build && cmake -GNinja -DCMAKE_TOOLCHAIN_FILE=${HOME}/toolchain -DFIPS=1 -DCMAKE_BUILD_TYPE=Release ..
+mkdir build && cd build && cmake -GNinja -DCMAKE_TOOLCHAIN_FILE=${HOME}/toolchain -DFIPS=1 -DCMAKE_BUILD_TYPE=Release -DCMAKE_C_FLAGS="-fPIC" -DCMAKE_CXX_FLAGS="-fPIC" ..
```

Comment:
@ggreenway my (ML) sources tell me that the FIPS spec is ambivalent about settings this so should be safe

not sure why its necessary from updating the llvm version but it seems to resolve the issue that it otherwise has

Reply:
The [document](https://csrc.nist.gov/CSRC/media/projects/cryptographic-module-validation-program/documents/security-policies/140sp4407.pdf) specifies a specific cmake command, and this deviates from it. 

I have no idea if that would break the requirements for FIPS. You'll need to get someone with deeper knowledge to sign off on this.

Reply:
From off-line discussion: https://github.com/golang/go/blob/master/src/crypto/internal/boring/build-boring.sh#L14 enables `-fPIC` in FIPS mode, so there is a precedent to enabling the flag.

Reply:
i have separate out this change here https://github.com/envoyproxy/envoy/pull/32901

the fact that it passes before/after upgrade seems to lend weight to the compliler becoming more picky theory, but not sure